ivy_tests/test_ivy/test_frontends/test_tensorflow/test_raw_ops.py

```diff
@@ -3058,3 +3058,96 @@ def test_tensorflow_Elu(
         features=x[0],
         name=name,
     )
+
+
+@st.composite
+def _conv2d_helper(draw):
+    input_dtype, \
+        x, \
+        filters, \
+        strides, \
+        dilations, \
+        data_format, \
+        padding = draw(_x_and_filters(
+        dtypes=helpers.get_dtypes("float", full=False),
+        data_format=st.sampled_from(["NHWC", "NCHW"]),
+        padding=st.sampled_from(["SAME", "VALID", "EXPLICIT"]),
+        type="2d",
+        dilation_min=1,
+        dilation_max=1,
+    ))
+
+    if padding == "EXPLICIT":
+        explicit_paddings = padding
+        pad_top = draw(st.integers(min_value=0, max_value=3))
+        pad_bottom = draw(st.integers(min_value=0, max_value=3))
+        pad_left = draw(st.integers(min_value=0, max_value=3))
+        pad_right = draw(st.integers(min_value=0, max_value=3))
+
+        if data_format == "NHWC":
+            explicit_paddings = [(0, 0),
+                                 (pad_top, pad_bottom),
+                                 (pad_left, pad_right),
+                                 (0, 0)]
+        else:
+            explicit_paddings = [(0, 0),
+                                 (0, 0),
+                                 (pad_top, pad_bottom),
+                                 (pad_left, pad_right)]
+
+    else:
+        explicit_paddings = []
+        padding = padding
+
+    return input_dtype, \
+        x, \
+        filters, \
+        dilations, \
+        data_format, \
+        strides, \
+        padding
+
+
+@handle_frontend_test(
+    fn_tree="tensorflow.raw_ops.Conv2D",
+    x_f_d_df=_conv2d_helper(),
+    test_with_out=st.just(False),
+)
+def test_tensorflow_Conv2D(
+    *,
+    x_f_d_df,
+    test_flags,
+    frontend,
+    fn_tree,
+    on_device,
+):
+    input_dtype, \
+        x, \
+        filters, \
+        dilation, \
+        data_format, \
+        stride, \
+        padding, \
+        explicit_paddings = x_f_d_df
+
+    # Broadcast strides and dilations to correct dims for the ground truth
+    # backend func to run correctly
+    stride = _convolution_broadcast_helper(
+        stride, num_spatial_dims=2, channel_index=3, name="strides"
+    )
+    dilation = _convolution_broadcast_helper(
+        dilation, num_spatial_dims=2, channel_index=3, name="dilations"
+    )
+    helpers.test_frontend_function(
+        input_dtypes=input_dtype,
+        test_flags=test_flags,
+        frontend=frontend,
+        fn_tree=fn_tree,
+        on_device=on_device,
+        input=x,
+        filter=filters,
+        strides=stride,
+        padding=padding,
+        data_format=data_format,
```

Comment:
you forgot to pass `explicit_paddings` here

Reply:
i made it same to the padding

Reply:
here I made it,` explicit_paddings = padding`

Reply:
then i perform the explicit_paddings

Reply:
@AnnaTz  I didn't understand it. 
in the test_tensorflow_Conv2D function, we should the take the value of explicit_paddings into a seperate variable and padding different variable

Reply:
and also return the explicit_paddings value in helpers.test_frontend_function

Reply:
> you get padding and explicit_paddings from x_f_d_df. You need to pass both separately

while running the test, it says expected 8 but got 7, i think that the x_f_d_df is not giving the value of explicit_padding or something